lib/matplotlib/backend_bases.py

```diff
@@ -220,18 +220,17 @@ def draw_path_collection(self, gc, master_transform, paths, all_transforms,
         recommended to use those generators, so that changes to the
         behavior of :meth:`draw_path_collection` can be made globally.
         """
-        path_ids = [
-            (path, transforms.Affine2D(transform))
-            for path, transform in self._iter_collection_raw_paths(
-                    master_transform, paths, all_transforms)]
+        path_ids = self._iter_collection_raw_paths(master_transform,
+                                                   paths, all_transforms)
 
         for xo, yo, path_id, gc0, rgbFace in self._iter_collection(
-                gc, master_transform, all_transforms, path_ids, offsets,
+                gc, master_transform, all_transforms, list(path_ids), offsets,
                 offsetTrans, facecolors, edgecolors, linewidths, linestyles,
                 antialiaseds, urls, offset_position):
             path, transform = path_id
-            transform = transforms.Affine2D(
-                            transform.get_matrix()).translate(xo, yo)
+            if xo != 0 or yo != 0:
+                transform = transform.frozen()
```

Comment:
Perhaps put a comment that the point here isn't really to freeze the transform, but just to copy it?  Alternatively use `transform + Affine2D().translate(xo, yo)`, which I guess works too.

Reply:
Added explanation